### PR TITLE
Tests and Fix #149: Getter returning Optional<T> of nullable T causing compilation error

### LINKED ISSUE #149: Getter returning Optional<T> of nullable T causing compilation error
Hi!

I'm having some troubles with class containing fields that are nullable and getters for those that return Optional<T>:

```
class Car {
    String someOptionalText;

    public Optional<String> getSomeOptionalText() {
        return Optional.ofNullable(someOptionalText);
    }
}
```

This causes the following error: 

> [ERROR] Line xx: Type mismatch: cannot convert from String to Optional&lt;String&gt;

Looking at the generated Serializer I see that it contains the following:

```
@Override
public Optional<String> getValue(Car bean, JsonSerializationContext ctx) {
    return getValueWithJsni(bean);
}

private native String getValueWithJsni(Car bean) /*-{
    return bean.@the.package.Car::someOptionalText;
}-*/;
```
Which of course isn't working. 

I've used the following config:

```
public class JacksonGWTConfig extends AbstractConfiguration {
    @Override
    protected void configure() {
        fieldVisibility(Visibility.ANY);
        getterVisibility(Visibility.NONE);
        isGetterVisibility(Visibility.NONE);
        setterVisibility(Visibility.NONE);
    }	
}
```
Looking in the code for method parse in PropertyParser I see that it uses the help method parseMethods, which doesn't care about the configuration. So it uses my getter that has a different return type (Optional) and later uses this as the return type for the Serializer.. 

Can this be solved with some config, is this a bug or is there something else I'm missing?

### PR DESCRIPTION
**Pull request by fr3driiik**
First PR ever. Let me know if I've missed anything :)

### CODE REVIEW

File: gwt-jackson/src/test/java/com/github/nmorel/gwtjackson/jackson/mapper/DifferentTypeGetterJacksonTest.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import org.junit.Test;
+
+import com.fasterxml.jackson.databind.JavaType;
+import com.github.nmorel.gwtjackson.jackson.AbstractJacksonTest;
+import com.github.nmorel.gwtjackson.shared.mapper.DifferentTypeGetterTester.DifferentTypeGetterBean;
+
+public class DifferentTypeGetterJacksonTest extends AbstractJacksonTest {
+
+	@Test
+	public void testDifferentTypeGetter()  {
```
Comment (nmorel) [inline gwt-jackson/src/test/java/com/github/nmorel/gwtjackson/jackson/mapper/DifferentTypeGetterJacksonTest.java]: Why don't you use `DifferentTypeGetterTester` ?

File: gwt-jackson/src/test/java/com/github/nmorel/gwtjackson/shared/mapper/DifferentTypeGetterTester.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	
+	public static DifferentTypeGetterTester INSTANCE = new DifferentTypeGetterTester();
+	
+	public void testDifferentTypeGetterBean(DifferentTypeGetterBeanMapper mapper) {
+		// This test is actually not that interesting. 
+		// Actual test is the generation of Serializer/Deserializer of DifferentTypeGetterBean.
+		String input = "{" +
+                "\"fieldWithOtherTypeAccessor\":\"true\"" +
+                "}";
+		DifferentTypeGetterBean bean = mapper.read(input);
```
Comment (nmorel) [inline gwt-jackson/src/test/java/com/github/nmorel/gwtjackson/shared/mapper/DifferentTypeGetterTester.java]: You should also test the write method and check that the result is also true and not "NO"

File: gwt-jackson/src/test/java/com/github/nmorel/gwtjackson/jackson/mapper/DifferentTypeGetterJacksonTest.java (hunk 1 above)
Comment (fr3driiik) [reply gwt-jackson/src/test/java/com/github/nmorel/gwtjackson/jackson/mapper/DifferentTypeGetterJacksonTest.java]: I only wanted to test the generation of serializer/deserializer, but I could as well have used the same tester. It is more consistent with the other tests also. Changing this :)

File: gwt-jackson/src/test/java/com/github/nmorel/gwtjackson/shared/mapper/DifferentTypeGetterTester.java (hunk 2 above)
Comment (fr3driiik) [reply gwt-jackson/src/test/java/com/github/nmorel/gwtjackson/shared/mapper/DifferentTypeGetterTester.java]: I've added more tests around this.

**Comment by fr3driiik** [comment]
The test has been updated :)

**Comment by nmorel** [comment]
Sorry for the delay and thanks for the update.  
I merged your change and I'm going to release it.

**Comment by fr3driiik** [comment]
No problems, thanks!

**Pull request merged**